optuna/storages/_rdb/storage.py

```diff
@@ -365,6 +378,12 @@ def get_all_study_summaries(self) -> List[StudySummary]:
                 except ValueError:
                     best_trial_frozen: Optional[FrozenTrial] = None
                 if best_trial:
+                    value = (
+                        session.query(models.TrialValueModel)
+                        .filter(models.TrialValueModel.trial_id == best_trial.trial_id)
+                        .filter(models.TrialValueModel.objective == 0)
+                        .one_or_none()
```

Comment:
[question] What is the criteria to define API in Model? Some query functionalities are implemented as API such as `where_study`, but others are not. For example, these lines are written by the `session.query(...)`.

Reply:
A possible criteria is that we define API in Model if we use the implementation more than once.

Reply:
Thank you for the important point. I will not touch `session.query()` directly in `storage.py`, but only in `models.py`. I'm currently touching `session.query()` in `storage.py,` which is inconsistent, so I'll fix it.

Reply:
The part I modified where `session.query()` is used has been rewritten in the method defined in `models.py`. I also rewrote the code parts that could be rewritten with the method in `models.py` without any obvious performance degradation. On the other hand, since it was not obvious how to rewrite it without performance degradation with `get_all_study_summaries` and `get_trials`, I did not rewrite it here. Whether the `session.query()` should only appear in `models.py` is a matter of debate.

Reply:
Just to fill in based on what I know, there's currently no real consensus whether queries should be defined via model methods or not. As a matter of fact, we previously discussed getting rid of all these methods to avoid confusion. I think it's reasonable given that many methods are quite short. If there are exceptions we could define functions in `storage.py`, keeping the consistency of the models being free from methods.

Reply:
Thanks for providing the information. It seems that whether we should remove the methods in `models.py` goes beyond the scope of this PR. It's good to discuss it later in another PR or issue.